Detailed description for enabling Windows nodes for AWS EKS (#727)
terraform-aws-modules/terraform-aws-eks: docs/faq.md

To enable Windows support for your EKS cluster, you should apply some configs manually. See the [Enabling Windows Support (Windows/MacOS/Linux)](https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html#enable-windows-support).

Windows worker nodes requires additional cluster role (eks:kube-proxy-windows). If you are adding windows workers to existing cluster, you should apply config-map-aws-auth again.

#### Algorithm for Windows nodes enabling:

> - Build AWS EKS cluster with the next workers configuration (default Linux):

Reviewer: ```suggestion
1. Create a worker group with with `platform = "linux"`:

```

```
worker_groups = [
    {
      name                          = "worker-group-linux"
      instance_type                 = "m5.large"
      platform                      = "linux"

>       additional_userdata           = "echo foo bar"

Reviewer: We don't need this, let's keep it simple.

```suggestion
```

      asg_desired_capacity          = 2

>       additional_security_group_ids = [///your_security_group]

Reviewer: We don't need this, let's keep it simple.

```suggestion
```

    },    
  ]
```

> - Apply commands from https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html#enable-windows-support (use tab with name `Windows`) 

Reviewer: ```suggestion

2. Apply commands from https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html#enable-windows-support (use tab with name `Windows`) 
```

> - Add one more worker group for Windows with required field `platform = "windows"` and update your cluster. Worker group example:

Reviewer: Is this correct? Or are you supposed to end up with 2 separate worker groups now? One linux and one windows?

```suggestion
3. Change worker group to `platform = "windows"` and run `terraform apply`:
```

```
{
      name                          = "worker-group-windows"
      instance_type                 = "m5.large"
      platform                      = "windows"
      asg_desired_capacity          = 1

>       additional_security_group_ids = [///your_security_group]      

Reviewer: We don't need this, let's keep it simple.

```suggestion
```

To enable Windows support for your EKS cluster, you should apply some configs manually. See the [Enabling Windows Support (Windows/MacOS/Linux)](https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html#enable-windows-support).

Windows worker nodes requires additional cluster role (eks:kube-proxy-windows). If you are adding windows workers to existing cluster, you should apply config-map-aws-auth again.

> #### Example configuration

Reviewer: Space after the head, like all other headings in this file

```suggestion
#### Example configuration

```

> 1. Build AWS EKS cluster with the next workers configuration (default Linux):

Reviewer: Let's add a note before the steps to explain why we still need linux here as this was confusing to me 😅

```suggestion
Amazon EKS clusters must contain one or more Linux worker nodes to run core system pods that only run on Linux, such as coredns and the VPC resource controller.

1. Build AWS EKS cluster with the next workers configuration (default Linux):

```

```
worker_groups = [
    {
      name                          = "worker-group-linux"
      instance_type                 = "m5.large"
      platform                      = "linux"
      asg_desired_capacity          = 2
    },    
  ]
```

2. Apply commands from https://docs.aws.amazon.com/eks/latest/userguide/windows-support.html#enable-windows-support (use tab with name `Windows`) 

> 3. Add one more worker group for Windows with required field `platform = "windows"` and update your cluster. Worker group example:

Reviewer: Add a blank line before code block

```suggestion
3. Add one more worker group for Windows with required field `platform = "windows"` and update your cluster. Worker group example:

```

```
worker_groups = [
    {
      name                          = "worker-group-linux"
      instance_type                 = "m5.large"
      platform                      = "linux"
      asg_desired_capacity          = 2
    },
    {
      name                          = "worker-group-windows"
      instance_type                 = "m5.large"
      platform                      = "windows"
      asg_desired_capacity          = 1
    },
  ]
```

> 4. After `kubectl get nodes` you can see cluster with mixed(Linux/Windows) nodes support.

Reviewer: Small grammar change

```suggestion
4. Wtih `kubectl get nodes` you can see cluster with mixed (Linux/Windows) nodes support.
```
